pkg/gofr/grpc.go

```diff
@@ -44,3 +45,21 @@ func (g *grpcServer) Run(c *container.Container) {
 		return
 	}
 }
+
+func (g *grpcServer) Shutdown(ctx context.Context) error {
+	return shutdownWithTimeout(ctx, func(ctx context.Context) error {
+		stopCh := make(chan struct{})
+		go func() {
+			g.server.GracefulStop()
+			close(stopCh)
+		}()
+
+		select {
+		case <-ctx.Done():
+			g.server.Stop() // Force stop if context is done
+			return ctx.Err()
+		case <-stopCh:
+			return nil
+		}
+	}, nil)
```

Comment:
I'm surprised here

We are using the shutdownWithTimeout but we have somehow a replicated code using what is in shutdownWithTimeout 

Am I missing something?

Reply:
Yes, @Umang01-hash please use the `shutdownWithTimeout` function, as we can pass `GracefulStop` in normal server stop and `Stop` for force stopping the gRPC server.

As for the time.After is concerned, I don't think that would be necessary as we are cancelling the context after the timeout, and we have `<- ctx.Done()` to catch those scenarios.

@Umang01-hash please also add comments on the specific functionality to better document the same!

Reply:
Passed `GracefulStop` and `Stop` for force stopping the gRPC server.

Also added the comments